src/Microsoft.OpenApi/Writers/OpenApiWriterAnyExtensions.cs

```diff
@@ -170,7 +171,14 @@ private static void WritePrimitive(this IOpenApiWriter writer, IOpenApiPrimitive
 
                 case PrimitiveType.Binary:
                     var binaryValue = (OpenApiBinary)primitive;
-                    writer.WriteValue(binaryValue.Value);
+                    if (binaryValue == null)
+                    {
+                        writer.WriteNull();
+                    }
+                    else
+                    {
+                        writer.WriteValue(Convert.ToBase64String(binaryValue.Value));
+                    }
                     break;
```

Comment:
@darrmi@microsoft.com  @saxu@microsoft.com 

I'm not sure if this is correct. 

- The spec says that "byte" type is base64 encoded characters. This means that our OpenApiByte which stores one byte is incorrect. Despite the singular name "byte", it seems to refer to multiple bytes. A reading from swagger.io also seems to suggest this: [https://swagger.io/docs/specification/data-models/data-types/#file](https://swagger.io/docs/specification/data-models/data-types/#file)

- Binary is defined as a sequence of octets, so it can't be Base64 string (since Base64 digit is 6-bit). I'm a bit at a loss at how to interpret this.

Reply:
@PerthCharern in my understand, from the spec it says binary is defined as a sequence of octets, while "byte" is used to represent one octet, meanwhile, one byte is base64 encoded characters (here "s" means one byte maybe more than one character base64 encode, so, binary is a sequence of base64 encoded characters. however, i leave it for @darrelmiller to give us more inputs.

Reply:
I really dislike the "format" property because it isn't well defined in the spec.  I do believe that format "byte" was intended to indicate a base64 encoded string.  I think the 'binary' format is basically saying that you should treat the UTF-8 code units as octets.  Not sure where that would be useful though.  

The "byte" format isn't representing a single byte, because if that were the case we would use the underlying integer type.

Reply:
I'll reach out to other folks to on the TSC to confirm.

Reply:
@darrelmiller & @PerthCharern Is there any update for this? Thanks.

Reply:
@xuzhg I got affirmation that a string that with a format binary should just be written out as a string, because what else could it be?  I agree it isn't very useful.

Reply:
@darrelmiller Ok. Let me merge that. @PerthCharern, if you have further concern, let's discuss offline. Thanks.

Reply:
@xuzhg Sorry, I guess I was not clear.  Format=Byte should be base64 encoded.  Format=Binary should just be serialized as a string.  Doesn't your change serialize format=binary as base64?

Reply:
@darrelmiller what kind of string for Format=Binary?

Reply:
@xuzhg I'm not sure I understand the question.  A UTF-8 string?  From an implmentation perspective, I think our reader/writer can basically ignore the format property when its value is binary.  If someone was then consuming that string value they would have the option of treating it like a array of chars and each char is a byte.

Reply:
@xuzhg Any chance you could help with creating a new PR that handles byte as base64 encoded and binary as just a simple string?

Reply:
@darrelmiller Sorry, I am not clear for my question. my original question is "how to convert a binary to/from a string in the OpenAPI.Net library?" 

Now, the underlying type of OpenApiBinary is "byte[]", if I create a OpenApiBinary to write, the library gives me the error. So, What's the built-in conversion in OpenApi.NET for our customers to use?

Reply:
My take would be:

- Store OpenApiBinary as byte[]. The reader converts the string into byte array assuming it knows each character is a byte. I think in C#, this can be done by `Encoding.ASCII.GetBytes(string);`. The writer does the reverse: `Encoding.ASCII.GetString(bytes);`

- We should also store OpenApiByte as byte[]. The reader converts the string into byte array using `Convert.FromBase64String`, and the writer converts the byte array into base64 string using `Convert.ToBase64String`.

This way we keep our principle of storing the underlying meaningful type that we does with Date/DateTime, and we allow the reader/writer to handle the (de-)serialization properly.

@xuzhg  @darrelmiller Thoughts?

Reply:
@PerthCharern 

So, do you think we will do like below:

1. Convert stringn to Binary?
`StringToBinary("fluxbytes.com")`

`// returns "01100110011011000111010101111000011000100111100101110100011001010111001100101110011000110110111101101101"`

2. Convert Binary string to string as below:

`BinaryToString("01100110011011000111010101111000011000100111100101110100011001010111001100101110011000110110111101101101");`

`// returns "fluxbytes.com"`

Reply:
I don't think so. I'm thinking of the logic that you already have in this PR, but with the right conversion function.

GetBytes instead of FromBase64String
GetString instead of ToBase64String